### PR TITLE
feat: popup general settings

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-4450

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/extension/popups/modals/SelectLanguageModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const lang = useLanguage()
+
+    const handleLanguageChange = useCallback(async (target: LanguageOptions) => {
+        await Services.Settings.setLanguage(target)
+    }, [])
+    const LANGUAGE_OPTIONS_MAP = useMemo(
+        () => ({
+            [LanguageOptions.__auto__]: t('popups_settings_language_auto'),
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/modals/SelectLanguageModal/index.tsx:53]: Use a resolver function stead.

Comment (nuanyang233) [reply packages/mask/src/extension/popups/modals/SelectLanguageModal/index.tsx:53]: But the i18n is required.

File: packages/shared-base/src/LegacySettings/settings.ts [hunk 2]
```diff
@@ -29,6 +29,21 @@ export const pluginIDsSettings = createGlobalSettings<Record<EnhanceableSite | E
     isEqual,
 )
 
+export const InjectSwitchSettings = createGlobalSettings<Record<EnhanceableSite, boolean>>(
```
Comment (Jack-Works) [inline packages/shared-base/src/LegacySettings/settings.ts]: You should use `createBulkSettings`.

File: packages/mask/src/utils/theme/usePopupTheme.ts [hunk 3]
```diff
@@ -1,14 +1,18 @@
 import { unstable_createMuiStrictModeTheme } from '@mui/material'
-import { MaskDarkTheme, MaskLightTheme } from '@masknet/theme'
+import { MaskDarkTheme, MaskLightTheme, useSystemPreferencePalette } from '@masknet/theme'
 import { languageSettings } from '@masknet/shared-base'
 import { useValueRef } from '@masknet/shared-base-ui'
 import { ThemeMode } from '@masknet/web3-shared-base'
-import { useThemeSettings } from '../../components/DataSource/useActivatedUI.js'
 import { useThemeLanguage } from './useThemeLanguage.js'
+import { useAppearance } from '../../../dashboard/pages/Personas/api.js'
```
Comment (Jack-Works) [inline packages/mask/src/utils/theme/usePopupTheme.ts]: you should move this hook to somewhere else, for example, `packages/mask/shared-ui`. you should not import from dashboard directly.

File: packages/mask/background/services/site-adaptors/connect.ts [hunk 4]
```diff
@@ -66,6 +66,21 @@ export async function getOriginsWithoutPermission(options: SitesQueryOptions = {
     return compact(await Promise.all(promises))
 }
 
+export async function getOriginsWithPermission() {
+    const groups = await getSupportedOrigins()
+    const promises = groups.map(async ({ origins, networkIdentifier }) => {
+        const unGrantedOrigins = compact(
```
Comment (Jack-Works) [inline packages/mask/background/services/site-adaptors/connect.ts]: originsWithNoPermission

File: packages/shared-base/src/LegacySettings/settings.ts (hunk 2 above)
Comment (nuanyang233) [reply packages/shared-base/src/LegacySettings/settings.ts]: Only the specified site has setting, key must be `EnhanceableSite`

**Review by guanbinrui — DISMISSED** [review]

File: packages/mask/background/services/site-adaptors/connect.ts (hunk 4 above)
Comment (Jack-Works) [reply packages/mask/background/services/site-adaptors/connect.ts]: please rename this. your word choosing is not natural

File: packages/shared-base/src/LegacySettings/settings.ts (hunk 2 above)
Comment (Jack-Works) [reply packages/shared-base/src/LegacySettings/settings.ts]: unfortunately, all other usages of `createBulkSettings` also limits key to `EnhanceableSite`. I hope we can keep consistent here.

File: packages/mask/src/extension/popups/pages/Settings/index.tsx [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+    const LANGUAGE_OPTIONS_MAP = useMemo(
+        () => ({
+            [LanguageOptions.__auto__]: t('popups_settings_language_auto'),
+            [LanguageOptions.enUS]: 'English',
+            [LanguageOptions.zhCN]: '简体中文',
+            [LanguageOptions.zhTW]: '繁体中文',
+            [LanguageOptions.jaJP]: '日本語',
+            [LanguageOptions.koKR]: '한국인',
+        }),
+        [],
```
Comment (Jack-Works) [inline packages/mask/src/extension/popups/pages/Settings/index.tsx]:
```suggestion
        [t],
```

File: packages/mask/src/extension/popups/pages/Settings/index.tsx [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+    )
+
+    const APPEARANCE_OPTIONS_MAP = useMemo(
+        () => ({
+            [Appearance.default]: t('popups_settings_appearance_default'),
+            [Appearance.light]: t('popups_settings_appearance_light'),
+            [Appearance.dark]: t('popups_settings_appearance_dark'),
+        }),
+        [],
```
Comment (Jack-Works) [inline packages/mask/src/extension/popups/pages/Settings/index.tsx]:
```suggestion
        [t],
```

File: packages/mask/src/extension/popups/pages/Settings/index.tsx [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+                                {t('popups_settings_data_correlation')}
+                            </Typography>
+                        </Box>
+                    </Box>
+                    <List className={classes.list}>
+                        <ListItem className={classes.listItem}>
+                            <ListItemText
+                                classes={itemClasses}
+                                primary={t('popups_settings_cloud_backup')}
+                                secondary="Not backed up"
```
Comment (Jack-Works) [inline packages/mask/src/extension/popups/pages/Settings/index.tsx]: no translation?

File: packages/mask/src/extension/popups/pages/Settings/index.tsx [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+                            />
+                            <Icons.ArrowRight size={24} />
+                        </ListItem>
+                        <ListItem className={classes.listItem}>
+                            <ListItemText
+                                classes={itemClasses}
+                                primary={t('popups_settings_local_backup')}
+                                secondary="Backup on 2022.12.20 12:23.20"
```
Comment (Jack-Works) [inline packages/mask/src/extension/popups/pages/Settings/index.tsx]: ? hard-coded data?

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/shared-base/src/LegacySettings/settings.ts (hunk 2 above)
Comment (Jack-Works) [reply packages/shared-base/src/LegacySettings/settings.ts]: not resolved, plz change!

**Review by guanbinrui — DISMISSED** [review]

File: packages/mask/src/extension/popups/modals/SelectAppearanceModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const mode = useAppearance()
+    const APPEARANCE_OPTIONS_MAP = useMemo(
+        () => ({
+            [Appearance.default]: t('popups_settings_appearance_default'),
+            [Appearance.light]: t('popups_settings_appearance_light'),
+            [Appearance.dark]: t('popups_settings_appearance_dark'),
+        }),
+        [],
```
Comment (Jack-Works) [inline packages/mask/src/extension/popups/modals/SelectAppearanceModal/index.tsx]:
```suggestion
        [t],
```

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**